File: surf-cloud-standalone/src/main/java/dev/slne/surf/cloud/standalone/test/TestPacketListener.java
package dev.slne.surf.cloud.standalone.test;


import dev.slne.surf.cloud.standalone.redis.RedisEventHandler;
import lombok.extern.flogger.Flogger;
import org.springframework.stereotype.Component;

@Component
@Flogger
public class TestPacketListener {

  @RedisEventHandler(TestPacket.CHANNEL)
  public void handle(TestPacket packet) {
    log.atInfo()
        .log("Received packet: %s", packet);
  }
}

File: surf-cloud-standalone/src/main/kotlin/dev/slne/surf/cloud/standalone/test/TestPacketListener.kt
package dev.slne.surf.cloud.standalone.test

import dev.slne.surf.cloud.api.common.TestPacket
import dev.slne.surf.cloud.api.common.meta.SurfNettyPacketHandler
import dev.slne.surf.surfapi.core.api.util.logger
import org.springframework.stereotype.Component


@Component
class TestPacketListener {
    private val log = logger()

    @SurfNettyPacketHandler
    fun onTestPacket(packet: TestPacket) {
        log.atInfo()
            .log("Received TestPacket: $packet")
    }
}
